src/client/pages/AssessmentSection/DataTable/Table/Row/RowData/Cell/hooks/useOnChange.ts

```diff
@@ -18,30 +19,47 @@ type Props = {
   data: TableData
   sectionName: string
 }
+export type OnChangeTaxon = (value: Taxon | string) => void
+export type OnChangeDefault = (
+  event: React.ChangeEvent<HTMLInputElement | HTMLSelectElement | HTMLTextAreaElement>
+) => void
+type OnChange = OnChangeTaxon | OnChangeDefault
```

Comment:
onChange should be generic. we should pass nodeValue here. we should refactor all other components

Reply:
Yes, let's do separate issue for this; I saw this but I think it is slightly more complex than scope of this PR

Reply:
I would like to start avoiding  committing to main branch code we already know we should change .
Is that really a big change ?

Reply:
The change actually can be minimum , without updating any existing components .

The hook we use to export the onChange function can export a 3rd function `onChangeNodeValue`.
The onChange will internally call onChangeNodeValue.

onChangeNodeValue accepts as param the NodeValue and it will be used only by Taxon component .

Would that work?

Reply:
Sounds good, I will update the component, only thing is I have to see this piece but should be straight forward
```
Cell.tsx

  return (
    // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
    <td className={className} onClick={showError} onKeyDown={showError}>
      <Component
        assessmentName={assessmentName}
        sectionName={sectionName}
        table={table}
        disabled={disabled || nodeValue?.odp}
        rowIndex={rowIndex}
        col={col}
        row={row}
        datum={datum}
        onChange={onChange} <------ 


Should we still call it onChange or should we add onChangeNodeValue and update Props?


        onPaste={onPaste}
      />
    </td>
```